File: FE/src/app/shared/models/user-dto.ts
import { RoleTypes } from '../enums/role-types';

export interface UserDto {
  id: number;
  fullName: string;
  genderType: boolean;
  dob: Date;
  avatar: string;
  email: string;
  phone: string;
  address: string;
  roleTypes: number[];
}

export interface UserDisplayDto {
  fullName: string;
  genderType: string;
  dob: Date;
  avatar: string;
  email: string;
  phone: string;
  address: string;
  roleTypes: string[];
}

export interface UserUpdateDto {
  fullName: string;
  genderType: boolean;
  dob: Date;
  avatar: string;
  phone: string;
  email: string;
  identityID: string;
  address: string;
  occupation: string;
  country: string;
  profileUrl: string;
}

File: Dreamy.Domain.Shared/Dtos/Auth/UserDto.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Dreamy.Domain.Shared.Dtos.Auth
{
    public class UserDto
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [StringLength(150)]
        public string FullName { get; set; }

        public bool? GenderType { get; set; }

        [Column(TypeName = "date")]
        public DateTime? Dob { get; set; }

        [StringLength(250)]
        public string? Avatar { get; set; }

        [StringLength(150)]
        public string Email { get; set; }

        [StringLength(50)]
        public string Phone { get; set; }

        [StringLength(250)]
        public string? Address { get; set; }
        public List<byte> RoleTypes { get; set; }
    }

    public class UserChangePassword
    {
        public string OldPassword { get; set; }
        public string NewPassword { get; set; }
    }
}
